src/HappyEyeBallsConnectionBuilder.php

```diff
@@ -83,36 +82,36 @@ public function connect()
             };
 
             $that->resolverPromises[Message::TYPE_AAAA] = $that->resolve(Message::TYPE_AAAA, $reject)->then($lookupResolve(Message::TYPE_AAAA));
-            $that->resolverPromises[Message::TYPE_A] = $that->resolve(Message::TYPE_A, $reject)->then(function (array $ips) use ($that, &$timer) {
+            $that->resolverPromises[Message::TYPE_A] = $that->resolve(Message::TYPE_A, $reject)->then(function (array $ips) use ($that) {
                 // happy path: IPv6 has resolved already (or could not resolve), continue with IPv4 addresses
                 if ($that->resolved[Message::TYPE_AAAA] === true || !$ips) {
                     return $ips;
                 }
 
                 // Otherwise delay processing IPv4 lookup until short timer passes or IPv6 resolves in the meantime
-                $deferred = new Promise\Deferred();
+                $deferred = new Promise\Deferred(function () use (&$ips) {
+                    // discard all IPv4 addresses if cancelled
```

Comment:
Doesn't this apply to both IPv4 and IPV6 addresses over time?

Reply:
See happy path above, so this timer will only run for 50ms if the `A` query resolves when the `AAAA` query is still pending. This is done to give a small edge for IPv6 as per https://tools.ietf.org/html/rfc8305#section-3. Consequently, this shouldn't affect any of the other scenarios, such as if IPv6 resolves first etc.

Reply:
Yeah fair point.